daemon/daemon_utils.go

```diff
@@ -52,3 +59,79 @@ func initialRuntime(baseDir string, runtimes map[string]types.Runtime) error {
 
 	return nil
 }
+
+// checkSnapshotter checks whether the given snapshotter is valid
+func checkSnapshotter(snapshotter string, containerdRootDir string) error {
+	var (
+		driverFound = false
+	)
+
+	fInfo, err := os.Stat(containerdRootDir)
+	if err != nil {
+		return fmt.Errorf("failed to stat containerd root dir: %s", err.Error())
+	}
+
+	if !fInfo.IsDir() {
+		return fmt.Errorf("containerd root dir %s is not a dicrectory", containerdRootDir)
+	}
+
+	infos, err := ioutil.ReadDir(containerdRootDir)
+	if err != nil {
+		return fmt.Errorf("failed to read dir %s:%s", containerdRootDir, err.Error())
```

Comment:
fmt.Errorf("failed to read dir %s: %v", containerdRootDir, err) ?

Reply:
Have Done it!